R/db-helpers.R

```diff
@@ -170,9 +187,14 @@ schema_postgres <- function(con, schema) {
   schema
 }
 
+ident_q <- function(...) {
+  # to avoid dependency on dbplyr define `ident_q()` here (not exported from dplyr)
+  structure(c(...), class = c("ident_q", "ident", "character"))
+}
+
 schema_mariadb <- function(con, schema) {
   if (is_null(schema)) {
-    schema <- sql("database()")
+    schema <- ident_q("database()")
```

Comment:
What's the advantage of using `ident_q()` here?

Reply:
It works better with `%in% !!...`:

``` r
library(dplyr, warn.conflicts = FALSE)
library(dbplyr, warn.conflicts = FALSE)
lf <- lazy_frame(x = 1)

lf %>% filter(x %in% !!sql("1"))
#> <SQL>
#> SELECT *
#> FROM `df`
#> WHERE (`x` IN 1)
lf %>% filter(x %in% !!ident_q("1"))
#> <SQL>
#> SELECT *
#> FROM `df`
#> WHERE (`x` IN (1))
```

<sup>Created on 2022-09-05 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>